### PR TITLE
APIMF-2324, APIMF-2323 and Raml Expression Parser refactor

### PR DESCRIPTION
**Pull request by tomsfernandez**
Related to: https://github.com/aml-org/amf-core/pull/104

### CODE REVIEW

File: amf-client/shared/src/test/resources/parser/array-type-expressions/base-type-array.expanded.jsonld [hunk 1]
```diff
@@ -264,36 +264,36 @@
...
                     "http://a.ml/vocabularies/document-source-maps#element": [
                       {
-                        "@value": "file://amf-client/shared/src/test/resources/parser/array-type-expressions/base-type-array.raml#/declarations/types/Person/link--2097854771"
+                        "@value": "file://amf-client/shared/src/test/resources/parser/array-type-expressions/base-type-array.raml#/declarations/types/Person/link-1304361918"
                       }
                     ],
                     "http://a.ml/vocabularies/document-source-maps#value": [
                       {
-                        "@value": "[(9,10)-(9,18)]"
+                        "@value": "Person[]"
```
Comment (llibarona) [inline amf-client/shared/src/test/resources/parser/array-type-expressions/base-type-array.expanded.jsonld]: este era el caso en el que me parece que deberiamos tener la info de Person (sin corchetes). Sabes si cambio?

File: amf-client/shared/src/test/scala/amf/RamlExpressionTests.scala [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  RamlExpressionLexer,
+  RamlExpressionParser,
+  Token
+}
+import amf.plugins.domain.shapes.models.{AnyShape, NodeShape, ScalarShape, UnionShape}
+import org.scalatest.FunSuite
+
+class RamlExpressionTests extends FunSuite {
+
+//  val declarationsFinder: DeclarationFinder = (name: String) => {
```
Comment (llibarona) [inline amf-client/shared/src/test/scala/amf/RamlExpressionTests.scala]: no se puede borrar y listo si no se va a usar mas?

Comment (tomsfernandez) [reply amf-client/shared/src/test/scala/amf/RamlExpressionTests.scala]: sepe

File: amf-client/shared/src/test/resources/parser/array-type-expressions/base-type-array.expanded.jsonld (hunk 1 above)
Comment (tomsfernandez) [reply amf-client/shared/src/test/resources/parser/array-type-expressions/base-type-array.expanded.jsonld]: Si, con esto cambio. Lo voy a tener que arreglar

Comment (tomsfernandez) [reply amf-client/shared/src/test/resources/parser/array-type-expressions/base-type-array.expanded.jsonld]: Solved with latest commit. The issue was caused by a hack in the Raml ArrayShapeParser. Couldn't undo the hack but improved the parser code to solve it.

**Review by looseale — APPROVED** [review]

**Pull request merged**